File: src/test/java/utils/bytes/XorTest.java
package utils.bytes;

import utils.codec.Hex;
import org.junit.Test;
import static org.junit.Assert.*;

public class XorTest {
    @Test
    public void sameLengthTest() {
        byte[] byte1 = Hex.fromHex("12345678");
        byte[] byte2 = Hex.fromHex("90abcdef");
        byte[] expected = Hex.fromHex("829f9b97");
        byte[] result = Xor.sameLength(byte1, byte2);
        assertArrayEquals(expected, result);
    }
}

File: src/test/kotlin/utils/bytes/XorTest.kt
package utils.bytes

import io.kotest.matchers.shouldBe
import org.junit.jupiter.api.Test
import utils.bytes.Xor.sameLength
import utils.codec.Hex.fromHex

@ExperimentalUnsignedTypes
class XorTest {
    @Test
    fun sameLengthTest() {
        val byte1 = fromHex("12345678")
        val byte2 = fromHex("90abcdef")
        val expected = fromHex("829f9b97")
        val result = sameLength(byte1, byte2)
        result shouldBe expected
    }
}
